From 255a354ec23064355079649c00bfcae6e8f2f0ba Mon Sep 17 00:00:00 2001
From: Pitchaiah Prakash <pitchaiah.p@variscite.com>
Date: Mon, 22 Apr 2024 19:36:50 +0200
Subject: [PATCH] imx8mm: Add dual bootloader support for imx8mm.

   1) Dual bootloader support mkimage has to adapt u-boot-lpddr4-ddr4-evk.itb changes.
   2) Add print_fit_hab_ddr4 to view memory sections on compiling.
   3) Change the order of deleting the dtb files.

Signed-off-by: Pitchaiah Prakash <pitchaiah.p@variscite.com>
---
 iMX8M/soc.mak | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 9d7556f..053d512 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -184,7 +184,6 @@ u-boot-lpddr4-ddr4-evk.itb: $(dtbs_lpddr4_ddr4_evk)
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs_lpddr4_ddr4_evk)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs_lpddr4_ddr4_evk) > u-boot-lpddr4-ddr4-evk.its
 	mkimage -E -p $(FIT_EXTERNAL_POSITION) -f u-boot-lpddr4-ddr4-evk.its u-boot-lpddr4-ddr4-evk.itb
-	@rm -f $(dtbs_lpddr4_ddr4_evk)
 
 dtb_ddr3l = valddr3l.dtb
 $(dtb_ddr3l):
@@ -346,12 +345,23 @@ flash_ddr4_evk_flexspi: $(MKIMG) u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb
 flash_lpddr4_ddr4_evk_no_hdmi: $(MKIMG) u-boot-spl-lpddr4-ddr4.bin u-boot-lpddr4-ddr4-evk.itb
 	./mkimage_imx8 -version $(VERSION) -fit -loader u-boot-spl-lpddr4-ddr4.bin $(SPL_LOAD_ADDR) -second_loader u-boot-lpddr4-ddr4-evk.itb 0x40200000 0x60000 -out $(OUTIMG)
 
+flash_lpddr4_ddr4_evk_no_hdmi_dual_bootloader: $(MKIMG) u-boot-spl-lpddr4-ddr4.bin u-boot-lpddr4-ddr4-evk.itb
+	./mkimage_imx8 -version $(VERSION) -fit -loader u-boot-spl-lpddr4-ddr4.bin $(SPL_LOAD_ADDR) -out $(OUTIMG)
+	./mkimage_imx8 -fit_ivt u-boot-lpddr4-ddr4-evk.itb 0x40200000 0x0 -out u-boot-ivt.itb
+
 flash_hdmi_spl_uboot: flash_evk
 
 flash_dp_spl_uboot: flash_dp_evk
 
 flash_spl_uboot: flash_evk_no_hdmi
 
+print_fit_hab_lpddr_ddr4: u-boot-nodtb.bin bl31.bin $(dtbs_lpddr4_ddr4_evk)
+	./$(PAD_IMAGE) tee.bin
+	./$(PAD_IMAGE) bl31.bin
+	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs_lpddr4_ddr4_evk)
+	FIT_DATA_POS=$(FIT_EXTERNAL_POSITION) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) VERSION=$(VERSION) ../$(SOC_DIR)/print_fit_hab.sh $(PRINT_FIT_HAB_OFFSET) $(dtbs_lpddr4_ddr4_evk)
+	@rm -f $(dtbs_lpddr4_ddr4_evk)
+
 print_fit_hab: u-boot-nodtb.bin bl31.bin $(dtb) $(supp_dtbs)
 	./$(PAD_IMAGE) $(TEE)
 	./$(PAD_IMAGE) bl31.bin
-- 
2.43.2

